pyccel/parser/syntactic.py

```diff
@@ -1030,7 +1030,7 @@ def _visit_ClassDef(self, stmt):
             if isinstance(visited_i, FunctionDef):
                 methods.append(visited_i)
                 visited_i.arguments[0].bound_argument = True
-            elif isinstance(visited_i, Pass):
+            elif isinstance(visited_i, (Pass, Comment)):
```

Comment:
This does not handle cases where a `CodeBlock` of comments is created like in the following:
```Python
class A:
    def __init__(self):
        self.x = 5
    #bad
    #bad
    #bad
    #bad
if __name__ == '__main__':
    a = A()
    print(a.x)
```

This would fix the problem:
```suggestion
            elif isinstance(visited_i, (Pass, Comment)):
                continue
            elif isinstance(visited_i, CodeBlock) and all(isinstance(x, Comment) for x in visited_i.body):
```

Reply:
Huh, I didn't realise that would be translated like that

Reply:
Fixed : c5952b929
Test added : 71588d45f